consensus/replay.go

```diff
@@ -278,6 +287,53 @@ func (h *Handshaker) Handshake(proxyApp proxy.AppConns) error {
 	return nil
 }
 
+// LocalSync restores stateStore and blockStore state which is not included in the snapshot
+// This function assumes that the implicit trusted local snapshot chunks have already been applied
+// State restoration depends on heights after the restored snapshot so the RPC dependency remains
+func (h *Handshaker) localSync(appBlockHeight uint64) (sm.State, *types.Commit, error) {
+	var state sm.State
+	var commit *types.Commit
+	var err error
+
+	if h.stateProvider == nil {
+		return sm.State{}, nil, errors.New("Missing State Provider")
+	}
+
+	pctx, pcancel := context.WithTimeout(context.TODO(), 30*time.Second)
+	defer pcancel()
+
+	// Optimistically build new state, so we don't discover any light client failures at the end.
+	state, err = h.stateProvider.State(pctx, appBlockHeight)
+	if err != nil {
+		h.logger.Error("failed to fetch and verify tendermint state", "err", err)
+		if err == light.ErrNoWitnesses {
+			return sm.State{}, nil, err
+		}
+		return sm.State{}, nil, errors.New("snapshot was rejected")
```

Comment:
Can we be more specific in this error message? It seems to me that what happened here was that Comet's light client verified the app hash present in the local snapshot of `app.db`, and this app hash was incorrect. Is this accurate?

Reply:
Referencing `func (s *lightClientStateProvider) State(ctx context.Context, height uint64) (sm.State, error) {`

The error conditions are failure to VerifyLightBlockAtHeight , RPC Client Creation Failure (Bad URL), and failure to fetch consensus parameters for the height.

It might be better to 1) return or 2) wrap the original error for the best accuracy.  Is there a preference?

Reply:
Yes, that's a good idea. Let's wrap the `err` to save the detail and return it to the caller.

Reply:
There is not snapshot involved here.

The error should say that the application state was rejected, instead.